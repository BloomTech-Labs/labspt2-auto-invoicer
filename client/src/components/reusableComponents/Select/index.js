import React from "react";

const Select = props => {
  return (
    <div className="form-group">
      <select
        name={props.name}
        value={props.selectedOption}
        onChange={props.controlFunc}
        className="form-select"
      >
        <option value="">{props.placeholder}</option>
        {props.options.map(opt => {
          return (
            <option key={opt} value={opt}>
              {opt}
            </option>
          );
        })}
      </select>
    </div>
  );
};

export default Select;
