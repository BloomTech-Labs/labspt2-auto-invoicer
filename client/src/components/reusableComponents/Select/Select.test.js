import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import Select from "../Select";

describe("<Select />", () => {
  it("renders Select", () => {
    const { asFragment } = render(
      <Select options={["test", "testing", "more testing"]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
