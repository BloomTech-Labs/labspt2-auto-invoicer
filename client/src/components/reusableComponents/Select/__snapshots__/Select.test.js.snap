// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders Select 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <select
      class="form-select"
    >
      <option
        value=""
      />
      <option
        value="test"
      >
        test
      </option>
      <option
        value="testing"
      >
        testing
      </option>
      <option
        value="more testing"
      >
        more testing
      </option>
    </select>
  </div>
</DocumentFragment>
`;
